src/LiveryBridge.ts

```diff
@@ -0,0 +1,296 @@
+  id: string;
+  type: 'event';
+  value: ValueType;
+}
+
+interface Deferred {
+  reject: (error: Error) => void;
+  resolve: (value: any) => void;
+}
+
+export class LiveryBridge {
+  protected deferredMap: Map<string, Deferred>;
+
+  protected handshakeId: string;
+
+  protected handshakePromise: Promise<void>;
+
+  protected listenerMap: Map<string, (value: any) => void>;
+
+  protected messageListener?: (message: string) => void;
+
+  protected targetWindow: Window;
+
+  protected targetWindowUrl: string;
+
+  protected version: string;
+
+  constructor(
+    targetWindow: Window,
+    targetWindowUrl: string,
+    version: string,
+    messageListener?: (message: string) => void,
+  ) {
+    this.deferredMap = new Map<string, Deferred>();
+    this.listenerMap = new Map<string, (value: any) => void>();
+
+    this.targetWindow = targetWindow;
+    this.targetWindowUrl = targetWindowUrl;
+    this.version = version;
+
+    this.messageListener = messageListener;
+
+    // Start listening for messages
+    window.addEventListener('message', (e) => {
+      if (LiveryBridge.isLiveryMessage(e.data)) {
+        this.handleMessage(e.data);
+      }
+    });
+
+    // Attempt to handshake.
+    this.handshakeId = uuid();
+    this.handshakePromise = this.sendHandshake(
+      this.handshakeId,
+      this.version,
+    ).catch((error: Error) => {
+      throw error;
+    });
+  }
+
+  static isCommandMessage(object: LiveryMessage): object is CommandMessage {
+    const command = object as CommandMessage;
+    return object.type === 'command' && !!object.id && !!command.name;
+  }
+
+  static isEventMessage(
+    object: LiveryMessage,
+  ): object is EventMessage<unknown> {
+    const event = object as EventMessage<unknown>;
+    return object.type === 'event' && !!event.id && !!event.value;
+  }
+
+  static isHandshakeMessage(object: LiveryMessage): object is HandshakeMessage {
+    return (
+      object.type === 'handshake' &&
+      !!(object as HandshakeMessage).version &&
+      !!object.id
+    );
+  }
+
+  static isLiveryMessage(object: any): object is LiveryMessage {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- Incoming object is any.
+    return !!object && object.isLivery === true && !!object.type;
+  }
+
+  static isRejectMessage(object: LiveryMessage): object is RejectMessage {
+    return (
+      object.type === 'reject' &&
+      !!object.id &&
+      !!(object as RejectMessage).error
+    );
+  }
+
+  static isResolveMessage<ValueType>(
+    object: LiveryMessage,
+  ): object is ResolveMessage<ValueType> {
+    return (
+      object.type === 'resolve' &&
+      !!object.id &&
+      !!(object as ResolveMessage<ValueType>).value
+    );
+  }
+
+  public sendCommand<ValueType>(
+    name: string,
+    id?: string,
+    arg?: string,
+  ): Promise<ValueType> {
+    return this.handshakePromise
+      .then(() => {
+        const commandId = id || uuid();
```

Comment:
As described above using default argument syntax is probably the best option here. Alternatively I'd suggest slightly improving this by using `??` instead of `||`.